internal/cloudsqlutil/engine.go

```diff
@@ -0,0 +1,130 @@
+package cloudsqlutil
```

Comment:
@kurtisvg thoughts on having engine in a shared package versus under the vector store implementation and any thoughts on sharing the engine between Alloydb and Cloud SQL?